pkg/event/truncate.go

```diff
@@ -0,0 +1,53 @@
+package event
+
+import (
+	"context"
+	"errors"
+	"sort"
+	"time"
+
+	internalv1 "github.com/acorn-io/runtime/pkg/apis/internal.acorn.io/v1"
+	"github.com/sirupsen/logrus"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Truncate truncates the set of stored account events to a maximum of limit once every period.
+func Truncate(ctx context.Context, client kclient.Client, period time.Duration, limit int) {
+	ticker := time.NewTicker(period)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			logrus.Debug("truncating eventinstances")
+			var events internalv1.EventInstanceList
+			if err := client.List(ctx, &events); err != nil {
+				logrus.WithError(err).Warn("failed to truncate events, error listing events")
+				continue
+			}
+
+			if len(events.Items) < limit {
+				// Nothing to do
+				continue
+			}
+
+			// Sort events by observed so that the newest events are first
+			sort.Slice(events.Items, func(i, j int) bool {
+				return events.Items[i].Observed.After(events.Items[j].Observed.Time)
+			})
+
+			var errs []error
+			for _, event := range events.Items[limit-1:] {
```

Comment:
super nit: if `limit` is 1000, I believe this will leave 999.

Reply:
I had a prof that would always say "what is a bit between friends"

Reply:
fixed